src/app/modules/main/profile_section/devices/module.nim

```diff
@@ -43,35 +43,43 @@ method load*(self: Module) =
 method isLoaded*(self: Module): bool =
   return self.moduleLoaded
 
+method getMyInstallationId(self: Module): string =
+  return self.controller.getMyInstallationId()
+
 proc isMyDevice(self: Module, installationId: string): bool =
   let myInstallationId = self.controller.getMyInstallationId()
   return installationId == myInstallationId
 
-proc initModel(self: Module) =
+method loadDevices*(self: Module) =
+  self.view.setDevicesLoading(true)
+  self.controller.asyncLoadDevices()
+  # TODO add error
+
+method onDevicesLoadingErrored*(self: Module) =
+  self.view.setDevicesLoading(false)
+  self.view.setDevicesLoadingError(true)
+
+method onDevicesLoaded*(self: Module, allDevices: seq[DeviceDto]) =
   var items: seq[Item]
-  let allDevices = self.controller.getAllDevices()
   for d in allDevices:
     let item = initItem(d.id, d.metadata.name, d.enabled, self.isMyDevice(d.id))
     items.add(item)
-
   self.view.model().addItems(items)
+  self.view.setDevicesLoading(false)
+  self.view.deviceSetupChanged()
 
 method viewDidLoad*(self: Module) =
-  self.initModel()
```

Comment:
Is that intended?

Reply:
Yes. We don't want to init the model from the app start, but rather when we change the section